internal/cache/cache_manager_test.go

```diff
@@ -499,11 +499,152 @@ func Test_persistCacheAndRestartCacheWithIt(t *testing.T) {
+							"false": testconvert.Interface(false),
+						},
+						DefaultRule: &flag.Rule{
+							VariationResult: testconvert.String("false"),
+						},
+					},
+				},
+			},
+		},
+		{
+			name:         "Empty initial flags",
+			initialFlags: map[string]dto.DTO{},
+			updatedFlags: map[string]dto.DTO{
+				"flag1": {
+					DTOv1: dto.DTOv1{
+						Variations: &map[string]*interface{}{
+							"true": testconvert.Interface(true),
+						},
+						DefaultRule: &flag.Rule{
+							VariationResult: testconvert.String("true"),
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "Remove a flag",
+			initialFlags: map[string]dto.DTO{
+				"flag1": {
+					DTOv1: dto.DTOv1{
+						Variations: &map[string]*interface{}{
+							"true": testconvert.Interface(true),
+						},
+						DefaultRule: &flag.Rule{
+							VariationResult: testconvert.String("true"),
+						},
+					},
+				},
+				"flag2": {
+					DTOv1: dto.DTOv1{
+						Variations: &map[string]*interface{}{
+							"false": testconvert.Interface(false),
+						},
+						DefaultRule: &flag.Rule{
+							VariationResult: testconvert.String("false"),
+						},
+					},
+				},
+			},
+			updatedFlags: map[string]dto.DTO{
+				"flag1": {
+					DTOv1: dto.DTOv1{
+						Variations: &map[string]*interface{}{
+							"true": testconvert.Interface(true),
+						},
+						DefaultRule: &flag.Rule{
+							VariationResult: testconvert.String("true"),
+						},
+					},
+				},
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// Test UpdateCache
+			mockNotifier := &mockNotificationService{}
+			cm := cache.New(mockNotifier, "", &fflog.FFLogger{LeveledLogger: slog.Default()})
+
+			err := cm.UpdateCache(tt.initialFlags, nil, false)
+			assert.NoError(t, err)
+
+			err = cm.UpdateCache(tt.updatedFlags, nil, false)
+			assert.NoError(t, err)
+			assert.Equal(t, 0, mockNotifier.notifyCalled, "Notify should not be called for UpdateCache")
+
+			flags, err := cm.AllFlags()
+			assert.NoError(t, err)
+			assert.Len(t, flags, len(tt.updatedFlags), "Cache should be updated with correct number of flags")
+
+			// Test UpdateCacheAndNotify
+			mockNotifier = &mockNotificationService{}
+			cm = cache.New(mockNotifier, "", &fflog.FFLogger{LeveledLogger: slog.Default()})
+
+			err = cm.UpdateCache(tt.initialFlags, nil, false)
+			assert.NoError(t, err)
+
+			err = cm.UpdateCache(tt.updatedFlags, nil, true)
+			assert.NoError(t, err)
+			assert.Equal(t, 1, mockNotifier.notifyCalled, "Notify should be called once for UpdateCacheAndNotify")
+
+			flags, err = cm.AllFlags()
+			assert.NoError(t, err)
+			assert.Len(t, flags, len(tt.updatedFlags), "Cache should be updated with correct number of flags")
+		})
+	}
+}
+
+type mockNotificationService struct {
+	notifyCalled int
+}
+
+func (m *mockNotificationService) Notify(oldFlags, newFlags map[string]flag.Flag, logger *fflog.FFLogger) {
+	m.notifyCalled++
+}
+
+func (m *mockNotificationService) Close() {}
```

Comment:
Same as the other test, I think we could use the already existing mock notifier available in the [`testutils/mock` directory](https://github.com/thomaspoignant/go-feature-flag/blob/main/testutils/mock/notifier_mock.go).

Reply:
Actually, the existing Notifier mock could not be used because:
* input parameter for `Notify()` are different
* doesn't implement `Close()` method.

However, I think it makes sense to move the mock to `testutils/mock` to be more consistent.
So I've created a separate mock in `testutils/mock` directory and used the new mock.